### PR TITLE
Add a batch of Coriolis worker server unit tests

### PR DESCRIPTION
**Pull request by smiclea**

### CODE REVIEW

File: coriolis/tests/worker/rpc/test_server.py [hunk 1]
```diff
@@ -192,3 +265,297 @@ def test_get_endpoint_instance(self, mock_get_provider, mock_get_secret, mock_va
...
+        mock_get_provider.return_value.validate_connection.assert_called_once_with(
+            mock.sentinel.context, mock_get_secret.return_value)
+
+        self.assertEquals(result, (True, None))
+
+        # handle SchemaValidationException
+        mock_validate.side_effect = exception.SchemaValidationException()
+        result = call_validate_endpoint_connection()
+        self.assertEquals(result, (False, (
+            "Schema validation for the provided connection parameters has "
```
Comment (Dany9966) [inline coriolis/tests/worker/rpc/test_server.py:521]: The tests should not care what error message gets passed. We should only assert whether `result[0]` is `False`. This also eliminates the requirement of double-editing both actual error message, and expected message in the test method (if someone would to edit the said error message).

File: coriolis/tests/worker/rpc/test_server.py [hunk 2]
```diff
@@ -192,3 +265,297 @@ def test_get_endpoint_instance(self, mock_get_provider, mock_get_secret, mock_va
...
+            "failed. Please ensure that you have included all the "
+            "necessary connection parameters and they are all properly "
+            "formatted for the '%s' Coriolis plugin in use." % (
+                mock.sentinel.platform_name))))
+
+        # handle ConnectionValidationException
+        mock_validate.side_effect = exception.ConnectionValidationException()
+        result = call_validate_endpoint_connection()
+        self.assertEquals(result, (False, 'An unknown exception occurred.'))
```
Comment (Dany9966) [inline coriolis/tests/worker/rpc/test_server.py:530]: Ditto.

File: coriolis/tests/worker/rpc/test_server.py [hunk 3]
```diff
@@ -192,3 +265,297 @@ def test_get_endpoint_instance(self, mock_get_provider, mock_get_secret, mock_va
...
+
+        # handle BaseException
+        mock_validate.side_effect = BaseException('test')
+        result = call_validate_endpoint_connection()
+        self.assertEquals(
+            result, (False, 'An unexpected connection validation exception ocurred: test'))
```
Comment (Dany9966) [inline coriolis/tests/worker/rpc/test_server.py:536]: Ditto. Please check and remove all other occurrences of exception message equal assertions.

**Review by Dany9966 — CHANGES_REQUESTED** [review]

File: coriolis/tests/worker/rpc/test_server.py [hunk 4]
```diff
@@ -192,3 +265,297 @@ def test_get_endpoint_instance(self, mock_get_provider, mock_get_secret, mock_va
...
+        (constants.PROVIDER_TYPE_ENDPOINT, 'connection_info_schema'),
+        (constants.PROVIDER_TYPE_REPLICA_IMPORT, 'destination_environment_schema'),
+        (constants.PROVIDER_TYPE_REPLICA_EXPORT, 'source_environment_schema'),
+        (constants.PROVIDER_TYPE_SOURCE_MINION_POOL,
+         'source_minion_pool_environment_schema'),
+        (constants.PROVIDER_TYPE_DESTINATION_MINION_POOL,
+         'destination_minion_pool_environment_schema'),
+        )
+    def test_get_provider_schemas(self, schema_type, mock_get_provider):
+        schemas = self.server.get_provider_schemas(
```
Comment (Dany9966) [inline coriolis/tests/worker/rpc/test_server.py:549]: Overrides imported `schemas` from outer-scope. Please rename it into something like `provider_schemas`

**Comment by smiclea** [comment]
All requests have been resolved. Also includes Black Formatter for code formatting.

File: coriolis/tests/worker/rpc/test_server.py [hunk 5]
```diff
@@ -119,76 +281,649 @@ def call_exec_task(report_to_conductor=True):
...
+        self.assertEqual(result[0], False)
+
+        # handle BaseException
+        mock_validate.side_effect = BaseException("test")
+        result = call_validate_endpoint_connection()
+        self.assertEqual(result[0], False)
+
+    @mock.patch.object(providers_factory, "get_provider")
+    @ddt.data(
+        (constants.PROVIDER_TYPE_ENDPOINT, "connection_info_schema"),
```
Comment (Dany9966) [inline coriolis/tests/worker/rpc/test_server.py:900]: Small nit, but you might as well have the same style of ddt data as the other longer ones. Try:

```python
    @ddt.data(
        (
            constants.PROVIDER_TYPE_ENDPOINT,
            "connection_info_schema"
        ),
        (
            constants.PROVIDER_TYPE_REPLICA_IMPORT,
            "destination_environment_schema",
        ),
        (
            constants.PROVIDER_TYPE_REPLICA_EXPORT,
            "source_environment_schema"
        ),
        (
            constants.PROVIDER_TYPE_SOURCE_MINION_POOL,
            "source_minion_pool_environment_schema",
        ),
        (
            constants.PROVIDER_TYPE_DESTINATION_MINION_POOL,
            "destination_minion_pool_environment_schema",
        ),
    )
```

**Review by Dany9966 — COMMENTED** [review]
Aside from the small nit, LGTM

File: coriolis/tests/worker/rpc/test_server.py (hunk 5 above)
Comment (smiclea) [reply coriolis/tests/worker/rpc/test_server.py]: Done

**Review by Dany9966 — APPROVED** [review]

**Pull request merged**